### PR TITLE
fix: slow typing 

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- Typing was slow

### Causes (Optional)

- Recomposition on each character input 
- Some unncessary recomposition of the UI 

### Solutions

- Annotate with @Stable classes that are using external module data to tell the compiler to favor smart composition with them
- Isolate the messge as the user is typing to MessageComposerInput only and send it on send button click to the parent 

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-950/1/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/common/bottomsheet/WireModalSheetLayout.kt [hunk 1]
```diff
@@ -60,7 +60,7 @@ fun MenuModalSheetLayout(
     sheetState: ModalBottomSheetState,
     coroutineScope: CoroutineScope,
     header: MenuModalSheetHeader = MenuModalSheetHeader.Gone,
-    menuItems: List<@Composable () -> Unit>,
+    menuItems: List<@Composable () -> Unit> ,
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/common/bottomsheet/WireModalSheetLayout.kt]: why the extra space here? 🤔

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt [hunk 2]
```diff
@@ -71,24 +68,14 @@ fun MessageComposer(
...
             messageComposerState = messageComposerState,
-            messageText = messageComposerState.messageText,
-            onMessageChanged = {
-                // we are setting it immediately in the UI first
-                messageComposerState.messageText = it
-                // we are hoisting the TextFieldValue text value up to the parent
-                if (messageText != it.text) {
-                    onMessageChanged(it.text)
-                }
+            onSendButtonClicked = {
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt]: Maybe wrapping lambdas in `remember` will also decrease unnecessary recompositions?

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageComposerViewModel.kt [hunk 3]
```diff
@@ -149,20 +148,15 @@ class MessageComposerViewModel @Inject constructor(
             showDeleteMessageDialog(messageToDeleteId, messageToDeleteIsSelf)
         }
         if (leftGroup || groupDeletedName != null) {
-            navigateBack(savedStateHandle.getBackNavArgs())
+            //  navigateBack(savedStateHandle.getBackNavArgs())
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageComposerViewModel.kt]: Should it be commented?

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationTopAppBar.kt [hunk 4]
```diff
@@ -47,8 +46,8 @@ fun ConversationScreenTopAppBar(
     onBackButtonClick: () -> Unit,
     onDropDownClick: () -> Unit,
     isDropDownEnabled: Boolean = false,
-    onSearchButtonClick: () -> Unit,
-    onPhoneButtonClick: () -> Unit,
+    onSearchButtonClick: () -> Unit = {},
+    onPhoneButtonClick: () -> Unit = {},
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationTopAppBar.kt]: Why the addition of default value?

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageComposerViewModel.kt (hunk 3 above)
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageComposerViewModel.kt]: If it should be kept as doing nothing then we can also remove the `if logic`

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/edit/EditMessageMenuItems.kt [hunk 5]
```diff
@@ -13,7 +13,7 @@ import com.wire.android.ui.common.bottomsheet.MenuItemIcon
 fun EditMessageMenuItems(
     isMyMessage: Boolean,
     onCopyMessage: () -> Unit,
-    onDeleteMessage: () -> Unit
+    onDeleteMessage: () -> Unit = {},
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/edit/EditMessageMenuItems.kt]: Here also, why the default value?

**Review by alexandreferris — COMMENTED** [review]
Great work! I just have a couple of comments

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-950/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4367441)(4367441)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4367441)(4367441)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt (hunk 2 above)
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt]: cant hurt :P

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-950/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-950/3/) **failed**.

**Review by alexandreferris — APPROVED** [review]
lgtm! 👌🏻

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-950/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4368168)(4368168)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4368168)(4368168)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-950/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4369901)(4369901)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4369901)(4369901)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-950/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4370195)(4370195)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4370195)(4370195)-dev-release.apk)

**Pull request merged**